var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -35,7 +36,13 @@ class Hpx(CMakePackage, CudaPackage):
 
     variant('networking', default=True,
             description='Support for networking and multi=node runs')
+    variant('mpi', default=False,
+            description='Enable the MPI parcelport (requires the networking'
+            ' variant to be on)')
+    conflicts('+mpi', when='~networking')
```

Comment:
Would it make sense to turn `networking` into a single valued variant (basically an enum) instead of adding a Boolean variant that depends on another one to be set?

Reply:
That would be a mutually exclusive set of values, no? When `networking` is on the default TCP parcelport is enabled. MPI is another one, and there is at least one other parcelport (libfabric) that could be added as well (I see there's a spack package for that, will look into that in a separate PR). They can all be enabled in the same build.

Reply:
> That would be a mutually exclusive set of values, no?

It can be both depending on use cases. Docs are [here](https://spack.readthedocs.io/en/latest/packaging_guide.html#multi-valued-variants) for you reference, but according to what you say I think you can get by with something similar to:
```python
variant(
    'networking',
    values=any_combination_of('tcp', 'libfabric', 'mpi'),
    description='...'
)
depends_on('mpi', when='networking=mpi')
```
Does that seem reasonable?

Reply:
Thanks and sorry for not checking the docs properly. That looks even better. I'll play around with this but that should describe the options pretty much perfectly.

Reply:
No worries, I know where to look as I've written those docs :laughing:

Reply:
Updated to use `any_combination_of`.